src/sink.rs

```diff
@@ -0,0 +1,58 @@
+use std::ffi::{CString, NulError};
+use std::ptr;
+
+use super::ffi::{self, c_uint};
+use super::types;
+
+#[derive(Debug)]
+pub struct Sink {
+    ptr: *mut ffi::Sink
+}
+
+unsafe impl Send for Sink {}
+
+// all non-Sync methods take &mut self:
+unsafe impl Sync for Sink {}
+
+#[derive(Debug)]
+pub enum SinkError {
+    NulInUri(NulError),
+    BadSink,
+}
+
+impl Sink {
+    pub fn open(uri: &str, sample_rate: usize) -> Result<Self, SinkError> {
+        let uri_cstr = CString::new(uri).map_err(SinkError::NulInUri)?;
+
+        let ptr = unsafe {
+            ffi::new_aubio_sink(uri_cstr.as_ptr(),
+                                ffi::uint(sample_rate))
+        };
+
+        if ptr == ptr::null_mut() {
+            Err(SinkError::BadSink)
+        } else {
+            Ok(Sink { ptr })
+        }
+    }
+
+    pub fn write(&mut self, write_data: &[types::Sample]) {
+        unsafe {
+            let input_fvec = ffi::fvec(write_data);
+            let write: c_uint = write_data.len() as u32;
+            {
+                ffi::aubio_sink_do(self.ptr, &input_fvec as *const ffi::FVec, write);
+            }
```

Comment:
Can you please remove the extra `{` `}` here?

Reply:
> Can you please remove the extra `{` `}` here?

Woops. Done!